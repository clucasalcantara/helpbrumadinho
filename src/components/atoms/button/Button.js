import React from 'react'
import PropTypes from 'prop-types'
import { Button as Btn } from 'semantic-ui-react'

const Button = ({ text, handleClick, className }) => (
  <Btn
    content={text}
    onClick={handleClick}
    className={`ui primary button ${className}`}
  />
)

Button.propTypes = {
  text: PropTypes.string.isRequired,
  handleClick: PropTypes.func,
  className: PropTypes.string
}

Button.defaultProps = {
  handleClick: () => {},
  className: ''
}

export default Button
