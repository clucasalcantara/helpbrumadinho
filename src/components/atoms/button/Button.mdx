---
name: Button
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import Button from './Button'

# Button component

## Basic usage

<Playground>
  <Button text="Click me" handleClick={() => alert('Clicked!')} />
</Playground>

## Properties

<PropsTable of={Button} />
