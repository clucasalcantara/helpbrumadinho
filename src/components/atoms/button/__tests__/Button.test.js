import React from 'react'
import { shallow } from 'enzyme'
import Button from '../Button'

describe('Button component', () => {
  it('should render without crash', () => {
    const wrapper = shallow(
      <Button text="Sugira agora" handleClick={() => console.log('clicked')} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
