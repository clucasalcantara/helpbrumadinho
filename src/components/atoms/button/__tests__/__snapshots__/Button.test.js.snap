// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className=""
    handleClick={[Function]}
    text="Sugira agora"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": "button",
      "className": "ui primary button ",
      "content": "Sugira agora",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "button",
        "className": "ui primary button ",
        "content": "Sugira agora",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
